content/blog/2020-10-tokio-0-3.md

```diff
@@ -0,0 +1,129 @@
+
+We have also refactored the runtime module. Tokio 0.2 exposed two types to interact with the Tokio runtime: `Runtime` and `Handle`. `Runtime` has a `block_on` method that required `&mut self`, whereas `Handle` has a `block_on` method that required `&self`. With Tokio 0.3, we collapsed `Runtime` and `Handle` into a single `Runtime`. Tokio 0.3’s `Runtime`‘s `block_on` now accepts `&self`, which supports concurrent calls to `block_on`. This also means `Runtime` implements `Send` and `Sync` and can therefore be stored in an `Arc`.
+
+An example configuring a multi-threaded runtime with 6 worker threads:
+
+```rust
+use tokio::runtime::Builder;
+
+let rt = Builder::new_multi_thread()
+    .worker_threads(6)
+    .enable_all()
+    .build()
+    .unwrap();
+
+rt.block_on(async {
+    println!("Hello world!");
+});
+```
+
+## Methods changed to `&self`
+
+We have overhauled how we handle wakers within our socket types to allow concurrent calls via `&self` rather than `&mut self`. To achieve this we had to rewrite how we handle wakers internally. The problem comes with how `poll_*` based methods work. Under the futures contract it states that a `poll_*` fn should only store the _last_ waker that it was called with. 
+
+> Note that on multiple calls to poll, only the Waker from the Context passed to the most recent call should be scheduled to receive a wakeup.
+
+From [`Future`](doc.rust-lang.org/std/future/trait.Future.html)
+
+This means that these `poll_*` can not be called from or woken concurrently. To make `&self` work for our `async fn` on our sockets types we refactored our io driver to use an internal instrusive linked-list to store the wakers. By doing this, it allows our socket types to accept `&self` rather than` &mut self` calls. 
+
+## Romoval of non-1.0 crates and future proofing for 1.0
+
+In our push for 1.0 we have removed many non-1.0 dependencies from the public API. This includes `bytes` and `mio`. This will allow us to push for a future proof and more stable 1.0 while we continue to innovate in our low-level crates.
+
+## Mio 0.7
+
+We have finally also upgraded `mio` to 0.7 which was originally released in an alpha in December of 2019 and did not make the cut for Tokio 0.2. Since then `mio` 0.7 has had time to mature and is finally making it into this release of Tokio 0.3. This upgrades some key dependencies like `winapi` from 0.2 to 0.3. Additionally, `mio` 0.7 now comes with a pure-rust implementation of [`wepoll`](https://github.com/piscisaureus/wepoll) which is a library that implements the `epoll` API for Windows based applications.
```

Comment:
```suggestion
We have finally also upgraded `mio` to 0.7, which was originally released in an alpha in December of 2019 and did not make the cut for Tokio 0.2. Since then `mio` 0.7 has had time to mature and is finally making it into this release of Tokio 0.3. This upgrades some key dependencies like `winapi` from 0.2 to 0.3. Additionally, `mio` 0.7 now comes with a pure-rust implementation of [`wepoll`](https://github.com/piscisaureus/wepoll) which is a library that implements the `epoll` API for Windows based applications.
```